### PR TITLE
GL-481 Added Green Lanes docs

### PR DESCRIPTION
**Pull request by jebw**
### Jira link

GL-481 

### What?

I have added/removed/altered:

- [x] Added Green Lanes documentation
 
### Why?

I am doing this because:

- So future developers have some explanation of how this works

### Deployment risks (optional)

- None - only docs changes

### CODE REVIEW

File: docs/green-lanes.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Windsor Framework - Green Lanes
+
+The green lanes support exists to allow categorisation of Goods Nomenclatures according to the Windsor Framework agreement.
```
Comment (alexdesi) [inline docs/green-lanes.md:3]: Simplification:

The Green Lanes support allows the categorisation ...

File: docs/green-lanes.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+We have a simple type table with each of the themes from Windsor Framework held in its own row.
+
+This provides both the category and a description of the Theme.
+
+### Category Assessments
+
+The categorisation itself is derived from the measures applied to a Goods Nomenclature. To categorise we need to match particular measures with the relevant Windsor Framework Theme.
+
+We have a join model - `CategoryAssessment` which matches particular measures on `measure_type_id`, `measure_generating_regulation_id` and `measure_generating_regulation_role`. This model the in turn points to the relevant `Theme` record which provides the category
```
Comment (alexdesi) [inline docs/green-lanes.md:21]: I'd avoid using "We have", and prefer a direct speech:

The model `CategoryAssessment` matches particular measures on ...
---
This model the in turn ...
something wrong in this phrase. Maybe "the" should be removed.
Also, what is "this model", does it refer to CategoryAssessment?

File: docs/green-lanes.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+This means there may potentially be multiple CategoryAssessments which apply to a particular GoodsNomenclature, and so multiple Themes with different Categories.
```
Comment (alexdesi) [inline docs/green-lanes.md:23]: the adverb "potentially" sounds superfluous.

File: docs/green-lanes.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+Our primary API is a generic GoodsNomenclature API which accepts any 6, 8 or 10 digit GN Item ID and returns the request GN, together with lists for its ancestors, its descendants, the CAs which apply directly to the GN and the CAs which apply to the Descendants.
+
+### Presented Category Assessments
+
+The 0 or more Category Assessments are included in the GN API - these are _presented_ to include additional relevant information which the API consumer can use to determine whether a particular CA does in fact apply in the API consumers context.
+
+Primarily this means a list of exemptions which the API consumer may determine are met and so be able to ignore a particular Category Assessment.
+
+_The effective category is determined by using the Theme from the restrictive category assessment which the API consumer determines does in fact apply. If the API consumer determines that no Category Assessments apply, then the Goods Nomenclature is determined to be **Standard Category**._
```
Comment (alexdesi) [inline docs/green-lanes.md:51]: A bit complex to understand (for me :) ) here: I got lost on "... which the API consumer determines does in fact apply."

File: docs/green-lanes.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+* To include the Exemptions list against the presented Category Assessment requires deriving the exemptions from the measures
```
Comment (alexdesi) [inline docs/green-lanes.md:53]: Is there any punctuation missing?  It's a bit hard to understand for my Latin brain.

File: docs/green-lanes.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+To achieve this the relevant measures are grouped together according to those with the same groups of exemptions (and geographical areas) and a separate CA is presented for each of these permutations.
+
+## Pseudo Measures and Pseudo Exemptions
+
+In addition using the Tariff data we also need to assign some category assessments independently of the data the Tariff measures may contain.
+
+### Pseudo Measures
+
+These are our own 'fake measures' which apply a Category Assessment directly to a GoodsNomenclature. They are a separate model (`GreenLanes::Measure`) and table which we control but internally the model is engineered to behaviour as close as possible to the tariff `Measure` model.
```
Comment (alexdesi) [inline docs/green-lanes.md:67]: Probably it needs a comma after "control" to separate the two clauses.
"to behaviour" sounds strange, probably it is "to behave"

File: docs/green-lanes.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+This means at a presenter level we can combine the pseudo measures with the regular measures. The do not affect the calculation of permuations of measures/exemptions because they apply to the whole world and so would be included in every permutation
```
Comment (alexdesi) [inline docs/green-lanes.md:69]: "The do not affect the calculation of permuations"  it's not clear :thinking: 
Misspelling permuations -> permutations
Missing comma at the end.

File: docs/green-lanes.md (hunk 6 above)
Comment (alexdesi) [inline docs/green-lanes.md:67]: Not sure what you mean with "our own  fake measures", :thinking:

File: docs/green-lanes.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+On the API they look identical to serialized tariff Measures but use an id of `GL0000XXX`.
+
+### Pseudo Exemptions
+
+These are our own 'fake exemptions' and they are applied directly to Category Assessments via their own join table.
```
Comment (alexdesi) [inline docs/green-lanes.md:75]: As before, I am not sure about the meaning of "our own"

**Review by alexdesi — COMMENTED** [review]
It's great that you are writing this! :rocket: 
This is very helpful.

The topic is of course complex. I would simplify the language:
Sometimes sentences are long with no punctuation.
There are some errors too.

File: docs/green-lanes.md (hunk 1 above)
Comment (jebw) [reply docs/green-lanes.md:3]: Problem is green lanes cannot explicitly categorise - it exists to provide data allowing the API client to categorise? I'll try and come up with a different phrasing

File: docs/green-lanes.md (hunk 6 above)
Comment (fasjkn) [reply docs/green-lanes.md:67]: Could we rephrase this please to : 

These are custom data points built by OTT to handle policies that don't explicitly appear in the EU tariff but are relevant under the Windsor Framework. They are designed to apply a Category Assessment directly to a Goods Nomenclature. These data points exist as a separate model (`GreenLanes::Measure`) and table that we control. Internally, this model is engineered to behave as similarly as possible to the tariff `Measure` model.

File: docs/green-lanes.md (hunk 1 above)
Comment (fasjkn) [reply docs/green-lanes.md:3]: Can we go with something like :

This API offers a capability for users to asses the categorization impact of Windsor Framework policies for past, current and future trades.

File: docs/green-lanes.md (hunk 6 above)
Comment (fasjkn) [reply docs/green-lanes.md:67]: Could we go with:

These are custom data points built by OTT to handle measures that don't explicitly appear in the tariff but are relevant under the Windsor Framework. They are designed to apply a Category Assessment directly to a Goods Nomenclature. These data points exist as a separate model (GreenLanes::Measure) and table that we control. Internally, this model is engineered to behave as similarly as possible to the tariff Measure model.

File: docs/green-lanes.md (hunk 8 above)
Comment (fasjkn) [reply docs/green-lanes.md:75]: Can we go with :

These are data points created by the OTT team and they applied directly to Category Assessments through their own join table. They behave similarly to regular exemptions, and account for gaps between the WF and the EU tariff data.

File: docs/green-lanes.md (hunk 1 above)
Comment (jebw) [reply docs/green-lanes.md:3]: Ended up with

| The green lanes support provides data allowing an API consumer to categorise Goods Nomenclatures according to the Windsor Framework agreement.

**Review by alexdesi — APPROVED** [review]
it looks good :100:

**Pull request merged**